include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,310 @@
+
+   // Query an extension.
+   // The returned pointer is owned by the provider.
+   // It is forbidden to call it before provider->init().
+   // You can call it within provider->init() call, and after.
+   const void *(CLAP_ABI *get_extension)(const struct clap_preset_discovery_provider *provider,
+                                         const char                                  *extension_id);
+} clap_preset_discovery_provider_t;
+
+// This interface isn't thread-safe
+typedef struct clap_preset_discovery_indexer {
+   clap_version_t clap_version; // initialized to CLAP_VERSION
+   const char    *name;         // eg: "Bitwig Studio"
+   const char    *vendor;       // eg: "Bitwig GmbH"
+   const char    *url;          // eg: "https://bitwig.com"
+   const char    *version;      // eg: "4.3", see plugin.h for advice on how to format the version
+
+   // Declares a preset filetype.
+   // Don't callback into the provider during this call.
+   void(CLAP_ABI *declare_filetype)(const struct clap_preset_discovery_indexer *indexer,
+                                    const clap_preset_discovery_filetype_t     *filetype);
+
+   // Declares a preset location.
+   // Don't callback into the provider during this call.
+   void(CLAP_ABI *declare_location)(const struct clap_preset_discovery_indexer *indexer,
+                                    const clap_preset_discovery_location_t     *location);
+
+   // Declares a preset collection.
+   // Don't callback into the provider during this call.
+   void(CLAP_ABI *declare_collection)(const struct clap_preset_discovery_indexer *indexer,
+                                      const clap_preset_discovery_collection_t   *collection);
+
+   // Sets the path to a watch file.
+   // Whenever the given file is "touched" (its modification time is updated),
+   // then the indexer shall invalidate all the data.
+   //
+   // This file shouldn't be touched when a new preset is added, the host shall detect the new
+   // preset using the OS features (file system monitoring).
+   //
+   // The invalidation file is useful if:
+   // - the set of filetypes changes
+   // - the set of locations changes
+   // - the set of collections changes
+   // - the metadata extraction code changes is located outside of the DSO containing this
+   //   preset provider
+   void(CLAP_ABI *set_invalidation_watch_file)(const struct clap_preset_discovery_indexer *indexer,
+                                               const char                                 *path);
+
+   // Query an extension.
+   // The returned pointer is owned by the indexer.
+   // It is forbidden to call it before provider->init().
+   // You can call it within provider->init() call, and after.
+   const void *(CLAP_ABI *get_extension)(const struct clap_preset_discovery_provider *provider,
+                                         const char                                  *extension_id);
+} clap_preset_indexer_t;
```

Comment:
Should be clap_preset_discovery_indexer_t I think?
Also appears in clap_preset_discovery_factory.create()

Reply:
Good catch again! :-)

Reply:
Haha, you've found a bug yourself, and fixed it :-)

I meant this:

line 277 should be:
} clap_preset_discovery_indexer_t;

line 304 should be:
const clap_preset_discovery_indexer_t                *indexer,

Reply:
Thank you again!